src/core/demangle.d

```diff
@@ -223,6 +223,12 @@ private struct Demangle
     }
 
 
+    char[] put(char c)
+    {
+        char[1] val = c;
+        return put(val[]);
```

Comment:
It's not `@safe` either way, the compiler just lacks the check for slicing the static array.